src/Php/PhpVersionResolver/ComposerJsonPhpVersionResolver.php

```diff
@@ -52,7 +52,7 @@ public static function resolve(string $composerJson): ?int
         $projectComposerJson = JsonFileSystem::readFilePath($composerJson);
 
         // give this one a priority, as more generic one
-        $requirePhpVersion = $projectComposerJson['require']['php'] ?? null;
+        $requirePhpVersion = $projectComposerJson['require']['php'] ?? $projectComposerJson['require']['php-64bit'] ?? null;
```

Comment:
Could you add link to https://github.com/composer/composer/issues/7914 here, so we have the context?

Reply:
Sure.